torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -389,6 +389,9 @@ def __init__(
         # if module is on CPU, use current device;
         self.compute_device = _get_default_cuda_device(module)
 
+        # Enum to indicate if we're in the forward/backward pass, idle, etc.
```

Comment:
Move the definition of `self.training_state` to before the first call to `named_parameters()` to avoid having to `getattr(self, "training_state", None)` in the overridden `named_parameters()`.